app/assets/javascripts/actions/story.js

```diff
@@ -218,6 +218,27 @@ export const updateCollapsedStory =
     );
   };
 
+export const expandOrCollapseStory =
+  (currentStory, from) =>
+  async (dispatch, _, { Story }) => {
+    if (!currentStory.collapsed)
+      return dispatch(toggleStory(currentStory.id, from));
+
+    dispatch(setLoadingStory(currentStory.id, from));
+    try {
+      const { data } = await projectStoriesService.fetchStory(currentStory);
+      dispatch(
+        updateStorySuccess(
+          { ...Story.deserialize(data.story), ...currentStory },
+          from
+        )
+      );
+      dispatch(toggleStory(currentStory.id, from));
+    } catch (error) {
+      dispatch(sendErrorNotification(error));
+    }
+  };
```

Comment:
I believe using `toggleStory` here might cause a problem. For example: what happens if the user clicks twice in the story? We'll dispatch two requests, and then we'll end up closing the story because toggle just inverts the open/closed state of the story. My suggestion:

Let's create two new actions (not thunks) called `expandStory` and `collapseStory` and use them to replace `toggleStory`. Then we rename `expandOrCollapseStory` to `toggleStory`, and then we check if story is loading before dispatching any request in this thunk:

```suggestion
export const toggleStory =
  (currentStory, from) =>
  async (dispatch, _, { Story }) => {
    if (!currentStory.collapsed) {
      return dispatch(collapseStory(currentStory.id, from));
    }
    
    if(isStoryLoading(currentStory)) { // we will need to create this new `isStoryLoading` in the models/beta/story.js file
      return; // we do nothing if it's loading
    }

    dispatch(setLoadingStory(currentStory.id, from));
    try {
      const { data } = await projectStoriesService.fetchStory(currentStory);
      dispatch(
        updateStorySuccess(
          { ...Story.deserialize(data.story), ...currentStory },
          from
        )
      );
      dispatch(expandStory(currentStory.id, from));
    } catch (error) {
      dispatch(sendErrorNotification(error));
    }
  };
```